src/gerbil/runtime/mop.ss

```diff
@@ -263,12 +267,15 @@ namespace: #f
                         precedence-list slot-vector slot-table properties constructor methods)
          :class)
       ;; this we know is a class
-      (:- (##structure class::t
-                       ;; gambit type fields
-                       type-id type-name type-flags type-super field-info
-                       ;; gerbil class fields
-                       precedence-list slot-vector slot-table properties constructor methods)
-          :class))))
+      (let (klass
+            (:- (##structure class::t
+                             ;; gambit type fields
+                             type-id type-name type-flags type-super field-info
+                             ;; gerbil class fields
+                             precedence-list slot-vector slot-table properties constructor methods)
+                :class))
+        (&class-type-precedence-list-set! klass (cons klass precedence-list))
```

Comment:
why?

Reply:
I kind of liked the differences between a 'class-type-precedence' and 'class-precedence' that we had before as it helped my mind to draw a distinction between meta- and introspection.